test/DynamoCoreTests/Models/OpenFileCommandTest.cs

```diff
@@ -116,12 +116,53 @@ public void TestInsertNodesCount()
             currentCount = this.CurrentDynamoModel.CurrentWorkspace.Nodes.Count();
             Assert.AreNotEqual(0, currentCount);
 
-            //If we try to insert the same graph, currently Dynamo should not allow us
-            // TODO - this part of the test might change if we include a logic
-            // to recreate the GUIDs of the 'same' nodes to allow for duplicate or partial duplicate insertions
+            //Assert 2x
             this.CurrentDynamoModel.InsertFileFromPath(wspath);
             var updatedCount = this.CurrentDynamoModel.CurrentWorkspace.Nodes.Count();
-            Assert.AreEqual(currentCount, updatedCount);
+            Assert.AreEqual(currentCount * 2, updatedCount);
+        }
+
+        [Test]
```

Comment:
do these tests cover ALL your new methods? It would be good to cover each new method you add with a unit test.

Reply:
There are 3 things to worry about, the way I see it:

Reply:
I will follow up on save as in a new PR and add test as well